Add Ipv6 Listeners

davama opened the pull request:
Awesome project! Thank you for publishing it

Simply took what you already have and added ipv6 settings.

Just in case someone asks, yes it works:
```
[2018-02-27T15:12:48,910][INFO ][logstash.pipeline        ] Starting pipeline {"id"=>"main", "pipeline.workers"=>2, "pipeline.batch.size"=>125, "pipeline.batch.delay"=>5, "pipeline.max_inflight"=>250}
[2018-02-27T15:12:49,109][INFO ][logstash.inputs.udp      ] Starting UDP listener {:address=>"10.xxx.xx.211:2055"}
[2018-02-27T15:12:49,111][INFO ][logstash.inputs.udp      ] Starting UDP listener {:address=>"10.xxx.xx.211:6343"}
[2018-02-27T15:12:49,113][INFO ][logstash.inputs.udp      ] Starting UDP listener {:address=>"10.xxx.xx.211:4739"}
[2018-02-27T15:12:49,114][INFO ][logstash.inputs.udp      ] Starting UDP listener {:address=>"[2xxx:xxxx:xxx:xx::211]:2055"}
[2018-02-27T15:12:49,115][INFO ][logstash.inputs.udp      ] Starting UDP listener {:address=>"[2xxx:xxxx:xxx:xx::211]:6343"}
[2018-02-27T15:12:49,117][INFO ][logstash.inputs.udp      ] Starting UDP listener {:address=>"[2xxx:xxxx:xxx:xx::211]:4739"}
[2018-02-27T15:12:49,117][INFO ][logstash.pipeline        ] Pipeline main started
[2018-02-27T15:12:49,132][INFO ][logstash.inputs.udp      ] UDP listener started {:address=>"10.xxx.xx.211:2055", :receive_buffer_bytes=>"106496", :queue_size=>"2000"}
[2018-02-27T15:12:49,133][INFO ][logstash.inputs.udp      ] UDP listener started {:address=>"[2xxx:xxxx:xxx:xx::211]:4739", :receive_buffer_bytes=>"106496", :queue_size=>"2000"}
[2018-02-27T15:12:49,134][INFO ][logstash.inputs.udp      ] UDP listener started {:address=>"[2xxx:xxxx:xxx:xx::211]:6343", :receive_buffer_bytes=>"106496", :queue_size=>"2000"}
[2018-02-27T15:12:49,135][INFO ][logstash.inputs.udp      ] UDP listener started {:address=>"[2xx:xxx:xxx:xx::211]:2055", :receive_buffer_bytes=>"106496", :queue_size=>"2000"}
[2018-02-27T15:12:49,136][INFO ][logstash.inputs.udp      ] UDP listener started {:address=>"10.xxx.xx.211:4739", :receive_buffer_bytes=>"106496", :queue_size=>"2000"}
[2018-02-27T15:12:49,137][INFO ][logstash.inputs.udp      ] UDP listener started {:address=>"10.xxx.xx.211:6343", :receive_buffer_bytes=>"106496", :queue_size=>"2000"}
[2018-02-27T15:12:49,204][INFO ][logstash.agent           ] Successfully started Logstash API endpoint {:port=>9600}
[2018-02-27T15:17:48,554][INFO ][logstash.filters.translate] refreshing dictionary file
```
Thought it would be useful.

-dave

robcowart commented:
Is the use-case here that you want to be able to listen on both IPv6 and IPv4 addresses on the same host simultaneously?

ruben-herold commented:
Yes on different ip's. Also be aware of this bug:

https://github.com/logstash-plugins/logstash-input-udp/pull/31

please manual update your input module ...

davama commented:
Sorry haven't replied

@robcowart yes, same host listening to both ipv4 and ipv6 in logstash

@ruben-herold yes, im aware of the bug see here: https://github.com/logstash-plugins/logstash-input-syslog/issues/42#issuecomment-374627298

davama commented:
Just to add....
ipv6 works on logstash 5.6.8 not on 6.2.2
I think it "working" is accidental

I had submitted a ticket in elasti community [site](https://discuss.elastic.co/t/logstash-input-fails-listening-ipv6/122114) but no traction

ruben-herold commented:
@davam: here it is now working on ipv6 only. It received the ipfix data from two Juniper MX240 via IPv6:

```
tcp6       0      0 :::2055                 :::*                    LISTEN      29292/java          
tcp6       0      0 127.0.0.1:9200          :::*                    LISTEN      29654/java          
tcp6       0      0 ::1:9200                :::*                    LISTEN      29654/java          
tcp6       0      0 127.0.0.1:9300          :::*                    LISTEN      29654/java          
tcp6       0      0 ::1:9300                :::*                    LISTEN      29654/java          
tcp6       0      0 127.0.0.1:9600          :::*                    LISTEN      29292/java          
udp6       0      0 :::2055                 :::*                                29292/java          
udp6       0      0 :::4739                 :::*                                29292/java          
udp6       0      0 :::6343                 :::*                                29292/java          
```

```
[root@ddosmon ~]# rpm -qa |grep logs
logstash-6.2.4-1.noarch
[root@ddosmon ~]# rpm -qa |grep elastic
elasticdump-2.2.0-2.el7.noarch
elastic-curator-3.2.3-1.el7.noarch
python-elasticsearch-1.9.0-1.el7.noarch
elasticsearch-6.2.4-1.noarch
[root@ddosmon ~]# 
```

davama commented:
Hmm...
let me test that out!
will report back
thank you @ruben-herold

davama commented:
:open_mouth: ipv6 it works!!! hurray!
Also noticed your v6 dashboard nice! :+1: 

```
[2018-04-24T13:49:05,235][INFO ][logstash.inputs.tcp      ] Starting tcp input listener {:address=>"10.x.x.213:4739", :ssl_enable=>"false"}
[2018-04-24T13:49:05,433][INFO ][logstash.inputs.tcp      ] Starting tcp input listener {:address=>"[xxxx::213]:4739", :ssl_enable=>"false"}
[2018-04-24T13:49:05,442][INFO ][logstash.inputs.udp      ] Starting UDP listener {:address=>"10.x.x.213:2055"}
[2018-04-24T13:49:05,461][INFO ][logstash.inputs.udp      ] Starting UDP listener {:address=>"10.x.x.213:6343"}
[2018-04-24T13:49:05,466][INFO ][logstash.inputs.udp      ] UDP listener started {:address=>"10.x.x.213:2055", :receive_buffer_bytes=>"106496", :queue_size=>"2000"}
[2018-04-24T13:49:05,470][INFO ][logstash.inputs.udp      ] UDP listener started {:address=>"10.x.x.213:6343", :receive_buffer_bytes=>"106496", :queue_size=>"2000"}
[2018-04-24T13:49:05,494][INFO ][logstash.inputs.udp      ] Starting UDP listener {:address=>"10.x.x.213:4739"}
[2018-04-24T13:49:05,495][INFO ][logstash.inputs.udp      ] UDP listener started {:address=>"10.x.x.213:4739", :receive_buffer_bytes=>"106496", :queue_size=>"2000"}
[2018-04-24T13:49:05,530][INFO ][logstash.inputs.udp      ] Starting UDP listener {:address=>"[xxxx::213]:2055"}
[2018-04-24T13:49:05,533][INFO ][logstash.inputs.udp      ] UDP listener started {:address=>"[xxxx::213]:2055", :receive_buffer_bytes=>"106496", :queue_size=>"2000"}
[2018-04-24T13:49:05,539][INFO ][logstash.inputs.udp      ] Starting UDP listener {:address=>"[xxxx::213]:6343"}
[2018-04-24T13:49:05,540][INFO ][logstash.inputs.udp      ] UDP listener started {:address=>"[xxxx::213]:6343", :receive_buffer_bytes=>"106496", :queue_size=>"2000"}
[2018-04-24T13:49:05,651][INFO ][logstash.inputs.udp      ] Starting UDP listener {:address=>"[xxxx::213]:4739"}
[2018-04-24T13:49:05,652][INFO ][logstash.inputs.udp      ] UDP listener started {:address=>"[xxxx::213]:4739", :receive_buffer_bytes=>"106496", :queue_size=>"2000"}
[2018-04-24T13:49:05,661][INFO ][logstash.pipeline        ] Pipeline started successfully {:pipeline_id=>"elastiflow", :thread=>"#<Thread:0x11cb7886@/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:247 sleep>"}
[2018-04-24T13:49:06,726][INFO ][logstash.pipeline        ] Starting pipeline {:pipeline_id=>"iperf3", "pipeline.workers"=>2, "pipeline.batch.size"=>125, "pipeline.batch.delay"=>50}
[2018-04-24T13:49:06,754][INFO ][logstash.outputs.elasticsearch] Elasticsearch pool URLs updated {:changes=>{:removed=>[], :added=>[http://elastic:xxxxxx@10.x.x.213:9200/]}}
[2018-04-24T13:49:06,755][INFO ][logstash.outputs.elasticsearch] Running health check to see if an Elasticsearch connection is working {:healthcheck_url=>http://elastic:xxxxxx@10.x.x.213:9200/, :path=>"/"}
[2018-04-24T13:49:06,801][WARN ][logstash.outputs.elasticsearch] Restored connection to ES instance {:url=>"http://elastic:xxxxxx@10.x.x.213:9200/"}
[2018-04-24T13:49:06,805][INFO ][logstash.outputs.elasticsearch] ES Output version determined {:es_version=>6}
```

```
[root@elastiflow-lab ~]# /usr/share/logstash/bin/logstash-plugin list --verbose | grep -E "flow|net"
logstash-codec-netflow (3.12.0)
logstash-codec-sflow (2.0.2)
[root@elastiflow-lab ~]# rpm -qa | grep -E "elas|logs"
logstash-6.2.4-1.noarch
elasticsearch-6.2.4-1.noarch
```

robcowart commented:
This was a good requirement, but I went a different direction on the implementation. This is now addressed in 3.0.0.
